### ISSUE TITLE
Fix unstable builds

### ISSUE THREAD

**Issue by martintmk** [App-vNext/Polly#1362]
We see quite a lot of failed builds. 

Either caused by code coverage check failing or mutation check issues:
https://github.com/App-vNext/Polly/actions/runs/5390349823/jobs/9785561789

Let's try to stabilize these:

- For mutation tests try to catch all "timeout" mutants. These can cause issues on different slower/faster machines.
- For code coverage, let's try to upload the report even if the check fails so we can diagnose where the coverage is missing.

**Comment by martintmk** [comment]
I think there is something wrong with coverlet reporting. For example, in the following build:
https://github.com/App-vNext/Polly/actions/runs/5532356778/jobs/10094322058

The console output complains about code coverage:

```
Calculating coverage result...
  Generating report 'D:\a\Polly\Polly\test\Polly.Core.Tests\coverage.net7.0.cobertura.xml'

+------------+--------+--------+--------+
| Module     | Line   | Branch | Method |
+------------+--------+--------+--------+
| Polly.Core | 99.86% | 99.79% | 100%   |
+------------+--------+--------+--------+

+---------+--------+--------+--------+
|         | Line   | Branch | Method |
+---------+--------+--------+--------+
| Total   | 99.86% | 99.79% | 100%   |
+---------+--------+--------+--------+
| Average | 99.86% | 99.79% | 100%   |
+---------+--------+--------+--------+

C:\Users\runneradmin\.nuget\packages\coverlet.msbuild\6.0.0\build\coverlet.msbuild.targets(72,5): error : The minimum line coverage is below the specified 100 [D:\a\Polly\Polly\test\Polly.Core.Tests\Polly.Core.Tests.csproj::TargetFramework=net7.0]
C:\Users\runneradmin\.nuget\packages\coverlet.msbuild\6.0.0\build\coverlet.msbuild.targets(72,5): error : The minimum branch coverage is below the specified 100 [D:\a\Polly\Polly\test\Polly.Core.Tests\Polly.Core.Tests.csproj::TargetFramework=net7.0]
C:\Users\runneradmin\.nuget\packages\coverlet.msbuild\6.0.0\build\coverlet.msbuild.targets(72,5): error :  [D:\a\Polly\Polly\test\Polly.Core.Tests\Polly.Core.Tests.csproj::TargetFramework=net7.0]
C:\Users\runneradmin\.nuget\packages\coverlet.msbuild\6.0.0\build\coverlet.msbuild.targets(72,5): error :    at Coverlet.MSbuild.Tasks.CoverageResultTask.Execute() in /_/src/coverlet.msbuild.tasks/CoverageResultTask.cs:line 250 [D:\a\Polly\Polly\test\Polly.Core.Tests\Polly.Core.Tests.csproj::TargetFramework=net7.0]
Test run for D:\a\Polly\Polly\test\Polly.Core.Tests\bin\Release\net6.0\Polly.Core.Tests.dll (.NETCoreApp,Version=v6.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.0-preview-23317-01+919ec8358820[228](https://github.com/App-vNext/Polly/actions/runs/5532356778/jobs/10094322058#step:5:229)cc5fa77ef000051c1d6875399 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.
```

While the report associated with the coverage says 100%:
<img width="1276" alt="image" src="https://github.com/App-vNext/Polly/assets/103487740/26ed326a-72a6-4a95-8fe3-faae5b465b37">

**Comment by martintmk** [comment]
Closing as I have not seen failure in a while.

**Comment by martincostello** [comment]
Looks like something slipped through the refactoring you did recently regarding the telemetry in the tests: https://github.com/App-vNext/Polly/actions/runs/5798416634/attempts/1

**Comment by martintmk** [comment]
> Looks like something slipped through the refactoring you did recently regarding the telemetry in the tests: https://github.com/App-vNext/Polly/actions/runs/5798416634/attempts/1

Should be fixed by #1466
